docs/datatypes/core/zio/error-management.md

```diff
@@ -0,0 +1,3124 @@
+2. While the second approach models the _divide by zero_ error by _dying_ the effect. We call these kinds of errors _unexpected errors_, _defects_ or _untyped errors_.
+
+We use the first method when we are handling errors as we expect them, and thus we know how to handle them. In contrast, the second method is used when we aren't expecting those errors in our domain, and we don't know how to handle them. Therefore, we use the _let it crash_ philosophy.
+
+In the second approach, we can see that the `divide` function indicates that it cannot fail because it's error channel is `Nothing`. However, it doesn't mean that this function hasn't any defects. ZIO defects are not typed, so they cannot be seen in type parameters.
+
+Note that to create an effect that will die, we shouldn't throw an exception inside the `ZIO.die` constructor, although it works. Instead, the idiomatic way of creating a dying effect is to provide a `Throwable` value into the `ZIO.die` constructor:
+
+```scala mdoc:compile-only
+import zio._
+
+val defect1 = ZIO.die(new ArithmeticException("divide by zero"))       // recommended
+val defect2 = ZIO.die(throw new ArithmeticException("divide by zero")) // not recommended 
+```
+
+Also, if we import a code that may throw an exception, all the exceptions will be translated to the ZIO defect:
+
+```scala mdoc:compile-only
+import zio._
+
+val defect3 = ZIO.succeed(throw new Exception("boom!"))
+```
+
+Therefore, in the second approach of the `divide` function, we do not require to manually die the effect in case of the _dividing by zero_, because the JVM itself throws an `ArithmeticException` when the denominator is zero. 
+
+When we import any code into the `ZIO` effect, any exception is thrown inside that code will be translated to _ZIO defects_ by default. So the following program is the same as the previous example:
+
+```scala mdoc:compile-only
+import zio._
+
+def divide(a: Int, b: Int): ZIO[Any, Nothing, Int] = 
+  ZIO.succeed(a / b)
+```
+
+Another important note is that if we `map`/`flatMap` a ZIO effect and then accidentally throw an exception inside the map operation, that exception will be translated to a ZIO defect:
+
+```scala mdoc:compile-only
+import zio._
+
+val defect4 = ZIO.succeed(???).map(_ => throw new Exception("Boom!"))
+val defect5 = ZIO.attempt(???).map(_ => throw new Exception("Boom!"))
+```
+
+### 3. Fatal Errors
+
+In ZIO, the `VirtualMachineError` and all its subtypes are the only errors considered fatal by the ZIO runtime. So if during the running application, the JVM throws any of these errors like `StackOverflowError`, the ZIO runtime considers it as a catastrophic fatal error. So it will interrupt the whole application immediately without safe resource interruption. None of the `ZIO#catchAll` and `ZIO#catchAllDefects` can catch these fatal errors. At most, if the `RuntimeConfig.reportFatal` is enabled, the application will log the stack trace before interrupting the whole application.
+
```

Comment:
I believe `RuntimeConfig` (and `RuntimeConfigAspect` have been deleted).

Reply:
I am making things challenging for @khajavi here. :smiley:

Reply:
No problem, I'm keeping the branch up to date regularly ;)

@adamgfraser 
Before the removal of `RuntimeConfig`, we had the ability to specify fatal errors:

```scala
val c: RuntimeConfig = ???
c.fatal { case x: Throwable => x.isInstanceOf[Foo] }
```

But I can't find the corresponding method in the new version. We won't support doing so anymore?